### PR TITLE
feat(3848): Allow users to enable prometheus feature flags

### PR DESCRIPTION
**Pull request by QuentinBisson**
Closes https://github.com/prometheus-operator/prometheus-operator/issues/3848

**Release Note Template (will be copied)**

```release-note:feature
Allow users to enable prometheus feature flags
```

cc @brancz

### CODE REVIEW

**Review by paulfantom — APPROVED** [review]
lgtm

File: pkg/apis/monitoring/v1/types.go [hunk 1]
```diff
@@ -185,6 +185,12 @@ type PrometheusSpec struct {
 	// ensure only clients authorized to perform these actions can do so.
 	// For more information see https://prometheus.io/docs/prometheus/latest/querying/api/#tsdb-admin-apis
 	EnableAdminAPI bool `json:"enableAdminAPI,omitempty"`
+	// Enable access to prometheus disabled features. By default, no features are enabled.
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1/types.go:188]:
```suggestion
	// Enable access to Prometheus disabled features. By default, no features are enabled.
```

File: pkg/prometheus/statefulset.go [hunk 2]
```diff
@@ -406,6 +406,10 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *operator.Config, shard in
 		promArgs = append(promArgs, "-web.enable-admin-api")
 	}
 
+	if p.Spec.EnableFeatures != nil && len(p.Spec.EnableFeatures) > 0 {
```
Comment (simonpasquier) [inline pkg/prometheus/statefulset.go:409]:
```suggestion
	if len(p.Spec.EnableFeatures) > 0 {
```

File: pkg/prometheus/statefulset_test.go [hunk 3]
```diff
@@ -1373,6 +1373,53 @@ func TestTerminationPolicy(t *testing.T) {
...
+		},
+	}, defaultTestConfig, nil, "", 0)
+
+	if err != nil {
+		t.Fatalf("Unexpected error while making StatefulSet: %v", err)
+	}
+
+	found := false
+	for _, flag := range sset.Spec.Template.Spec.Containers[0].Args {
+		t.Log(flag)
```
Comment (simonpasquier) [inline pkg/prometheus/statefulset_test.go:1389]:
```suggestion
```

**Review by simonpasquier — COMMENTED** [review]
lgtm, only a few stylistic comments

**Comment by QuentinBisson** [comment]
Thanks @simonpasquier for the comments :)

**Review by simonpasquier — APPROVED** [review]
thanks!

**Pull request merged**